### ISSUE TITLE
commitlog consumes all of the disk space and cause service down

### ISSUE THREAD

**Issue by JerryKwan** [scylladb/scylladb#11988]
This is Scylla's bug tracker, to be used for reporting bugs only.
If you have a question about Scylla, and not a bug, please ask it in
our mailing-list at scylladb-dev@googlegroups.com or in our slack channel.

- [x] I have read the disclaimer above, and I am reporting a suspected malfunction in Scylla.

*Installation details*
Scylla version (or git commit hash): scylla-jmx-5.0.5-0.20221009.5a97a1060.noarch
Cluster size: 10 nodes
OS (RHEL/CentOS/Ubuntu/AWS AMI): CentOS

Seems like the version of v5.0.5 does not respect the config of commitlog_total_space_in_mb: -1, and will consume all of the disk space and cause the service down.
The total memory is 128G, but after running for some days the commitlog directory occupies almost 392G. As some documents said it should have some limit regarding the memory size

**Comment by elcallio** [comment]
See #2091 and similar. In 5.0.x, commitlog hard limit is not on by default. Use `commitlog_use_hard_size_limit: True` to disallow unbounded size growth.

**Comment by JerryKwan** [comment]
@elcallio Sorry, I am not sure why https://github.com/scylladb/scylladb/issues/2091 is related to this issue

**Comment by elcallio** [comment]
@JerryKwan Yes, use `commitlog_use_hard_size_limit: True`. This will enforce limiting disk footprint ("normal" operation tries to restrain, but will prio throughput). 

Remember however this will in turn cause certain latencies as the system will have to wait for data to become fully persisted (written to sstable) before it can make commitlog space available to handle new writes. If you have a workload that today causes unbounded commitlog size, chances are you will experience lowered throughput. 

`commitlog_total_space_in_mb: -1` only means that disk footprint limits are calculated automatically based on system RAM memory. You might actually be better served by setting it to whatever size you are prepared to give it.

**Comment by JerryKwan** [comment]
@elcallio I have add commitlog_use_hard_size_limit: true, but it still consumes much more disk space than the system RAM memory.  It still consumes almost all of the disk space. Any suggestions?

**Comment by JerryKwan** [comment]
@elcallio Any suggestions why commitlog_use_hard_size_limit: true does not work? It still exceed the setting of commitlog_total_space_in_mb. And What should I do to limit the total size of commit log use?

**Comment by elcallio** [comment]
@JerryKwan If you still get unbounded CL size growth you need to provide more info. Can you turn up the log level for `commitlog` to debug and provide the logs? Also your complete settings + HW.
